crates/iroha_executor/src/default/mod.rs

```diff
@@ -1235,40 +1231,49 @@ pub mod role {
         isi: &Register<Role>,
     ) {
         let role = isi.object();
+        let grant_role = &Grant::account_role(role.id().clone(), role.grant_to().clone());
         let mut new_role = Role::new(role.id().clone(), role.grant_to().clone());
 
         // Exception for multisig roles
-        let mut is_multisig_role = false;
-        if let Some(tail) = role
-            .id()
-            .name()
-            .as_ref()
-            .strip_prefix("multisig_signatory_")
         {
-            let Ok(account_id) = tail.replacen('_', "@", 1).parse::<AccountId>() else {
-                deny!(executor, "Violates multisig role format")
-            };
-            if crate::permission::domain::is_domain_owner(
-                account_id.domain(),
-                &executor.context().authority,
-                executor.host(),
-            )
-            .unwrap_or_default()
-            {
-                // Bind this role to this permission here, regardless of the given contains
-                let permission = CanExecuteTrigger {
-                    trigger: format!(
-                        "multisig_transactions_{}_{}",
-                        account_id.signatory(),
-                        account_id.domain()
+            use crate::permission::domain::is_domain_owner;
+
+            const DELIMITER: char = '/';
+            const MULTISIG_SIGNATORY: &str = "MULTISIG_SIGNATORY";
+
+            fn multisig_account_from(role: &RoleId) -> Option<AccountId> {
+                role.name()
+                    .as_ref()
+                    .strip_prefix(MULTISIG_SIGNATORY)?
+                    .rsplit_once(DELIMITER)
+                    .and_then(|(init, last)| {
+                        format!("{last}@{}", init.trim_matches(DELIMITER))
+                            .parse()
+                            .ok()
+                    })
+            }
+
+            if role.id().name().as_ref().starts_with(MULTISIG_SIGNATORY) {
+                if let Some(multisig_account) = multisig_account_from(role.id()) {
```

Comment:
Would 
```rust
let Some(multisig_account) = multisig_account_from(role.id()) else { 
    deny!(executor, "violates multisig role name format")
}
```
work here? I find it when the shorter case of the if-else comes first it's easier to read.